juju/juju
# fix: broken authorized keys bootstrap test

This commit fixes broken authorized keys test for bootstrap. It has been a problem that the authorized_keys bootstrap tests have been failing in our CI environment and not on local machines when testing.

The reason for this failaure is due to our CI environment having set the JUJU_DATA env variable. This set of tests needs to be able to hijack the Juju home data directory to establish a controlled environment.

Because the CI system had this env set Juju was using the wrong directory for it's data. The fix is to make sure that we set this variable for the duration of the test to the value we want.

## Checklist

- ~[] Go unit tests, with comments saying what you're testing~

## QA steps

We can confirm this fix with the following steps:

1. cd tests
2. `JUJU_DATE=foobar ./main.sh -v authorized_keys test_bootstrap_authorized_keys`

If the tests run successfully then everything is resolved.

## Documentation changes

**Jira card:** JUJU-6889
